validator/keymanager/imported/delete.go

```diff
@@ -0,0 +1,71 @@
+package imported
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+	ethpbservice "github.com/prysmaticlabs/prysm/proto/eth/service"
+	"github.com/prysmaticlabs/prysm/validator/accounts/petnames"
+	"github.com/sirupsen/logrus"
+)
+
+// DeleteKeystores takes in public keys and removes the accounts from the wallet.
+// This includes their disk keystore and cached keystore, but maintains the slashing
+// protection history in the database.
+func (km *Keymanager) DeleteKeystores(
+	ctx context.Context, publicKeys [][]byte,
+) ([]*ethpbservice.DeletedKeystoreStatus, error) {
+	statuses := make([]*ethpbservice.DeletedKeystoreStatus, len(publicKeys))
+	for i, publicKey := range publicKeys {
+		var index int
+		var found bool
+		for j, pubKey := range km.accountsStore.PublicKeys {
+			if bytes.Equal(pubKey, publicKey) {
+				index = j
+				found = true
+				break
+			}
+		}
+		if !found {
+			statuses[i] = &ethpbservice.DeletedKeystoreStatus{
```

Comment:
is this transactional? i'm wondering if we should validate first checkign for duplicates and such before starting any of this delete process

Reply:
> _we decided to make it transactional on  a per-keystore basis, as that maps more naturally to deleting individual keystores from disk. If Prysm has a single database of keys that it can modify atomically for all the deleted keystores then that’s awesome, it can use a single transaction for the whole request. What it shouldn’t do is fail the request if some keystores fail and others succeed. It should return “deleted” statuses for all the successes and “error” for the others, or if the successes can be reverted it could return “error” for all keystores (but still a 200 OK at the HTTP level)_
nvm light house response on this doesnt need to be transactional